### PR TITLE
Feature/swapper performance logging

### PR DESCRIPTION
**Pull request by tznind**
## Proposed Changes

Summarise your changes, and include a link to the [CHANGELOG](https://github.com/SMI/SmiServices/blob/master/CHANGELOG.md) diff.

## Types of changes

What types of changes does your code introduce? Tick all that apply.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation-Only Update (if none of the other choices apply)
  - In this case, ensure that the message of the head commit from the source branch is prefixed with `[skip ci]`

## Checklist

By opening this PR, I confirm that I have:

- [x] Reviewed the [contributing](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTING.md) guidelines for this repository
- [x] Ensured that the PR branch is in sync with the target branch (i.e. it is automatically merge-able)
- [ ] Updated any relevant API documentation
- [x] Created or updated any tests if relevant
- [x] Accurately updated the [CHANGELOG](https://github.com/SMI/SmiServices/blob/master/CHANGELOG.md)
  - NOTE: This ***must*** include any changes to any of the following files: default.yaml, any of the RabbitMQ server configurations, GlobalOptions.cs
- [ ] Listed myself in the [CONTRIBUTORS](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTORS.md) file 🚀
- [x] Requested a review by one of the repository maintainers

## Issues

If relevant, tag any issues that are *expected* to be resolved with this PR. E.g.:

- Closes #\<issue-number>
- ...

### CODE REVIEW

File: src/microservices/Microservices.IdentifierMapper/Execution/Swappers/ForGuidIdentifierSwapper.cs [hunk 1]
```diff
@@ -37,29 +38,36 @@ public ForGuidIdentifierSwapper()
...
+            using(new TimeTracker(DatabaseStopwatch))
+                CreateTableIfNotExists();
         }
 
-        public string GetSubstitutionFor(string toSwap, out string reason)
+        public override string GetSubstitutionFor(string toSwap, out string reason)
         {
             reason = null;
 
             if (toSwap.Length > 10)
```
Comment (rkm) [inline src/microservices/Microservices.IdentifierMapper/Execution/Swappers/ForGuidIdentifierSwapper.cs]: Just noticed this. Looks like this specifically looks for CHIs (length 10), but some of the invalid PatientIDs we want to create a GUID for won't pass this check>

**Comment by rkm** [comment]
Looks fine other than that one comment

File: src/microservices/Microservices.IdentifierMapper/Execution/Swappers/ForGuidIdentifierSwapper.cs (hunk 1 above)
Comment (tznind) [reply src/microservices/Microservices.IdentifierMapper/Execution/Swappers/ForGuidIdentifierSwapper.cs]: Ok I'll check in a fix for this

Comment (tznind) [reply src/microservices/Microservices.IdentifierMapper/Execution/Swappers/ForGuidIdentifierSwapper.cs]: I have committed a fix for this, it now looks at the length of the column and enforces that as the maximum.

**Pull request merged**